buildSrc/src/main/java/com/google/firebase/gradle/plugins/Changelog.kt

```diff
@@ -0,0 +1,337 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.google.firebase.gradle.plugins
+
+import java.io.File
+
+// TODO(b/281873509) - Add documentation for CHANGELOG.md
+/**
+ * Type safe representation of `CHANGELOG.md` files.
+ *
+ * For more information on what a `CHANGELOG.md` file entails, see the following documentation:
+ * [Changelog Files](#)
+ *
+ * @property releases previous releases and the data associated with them
+ * @see fromFile
+ * @see fromString
+ */
+data class Changelog(val releases: List<ReleaseEntry>) {
+  companion object {
+    /**
+     * Regex for finding the version titles in a changelog file.
+     *
+     * The regex can be described as such:
+     * - Find lines that start with `#`, followed by a space and any amount of characters
+     *
+     * For example:
+     * ```markdown
+     * # Unreleased
+     *
+     * # 18.3.7
+     * [feature] Added collection of version control information generated by the
+     *   Android Gradle plugin (AGP).
+     *
+     * # 18.3.6
+     * [feature] Added support for upcoming [crashlytics] features to report
+     *   GWP-ASan crashes on supported API levels.
+     * ```
+     *
+     * Will find the following groups:
+     * ```kotlin
+     * [
+     *   "Unreleased",
+     *   "18.3.7",
+     *   "18.3.6"
+     * ]
+     * ```
+     */
+    val RELEASE_VERSION_REGEX = Regex("^# (.+)", RegexOption.MULTILINE)
+
+    /**
+     * Parses a [Changelog] from a [String].
+     *
+     * For example, the following is a valid string:
+     * ```markdown
+     * # Unreleased
+     *
+     * # 18.3.7
+     * [feature] Added collection of version control information generated by the
+     *   Android Gradle plugin (AGP).
+     *
+     * ## Kotlin
+     * The Kotlin extensions library transitively includes the updated
+     * `firebase-crashlytics` library. The Kotlin extensions library has no additional
+     * updates.
+     *
+     * # 18.3.6
+     * [feature] Added support for upcoming [crashlytics] features to report
+     *   GWP-ASan crashes on supported API levels.
+     * ```
+     *
+     * @see RELEASE_VERSION_REGEX
+     * @see ReleaseEntry
+     */
+    fun fromString(string: String): Changelog {
+      val content =
+        RELEASE_VERSION_REGEX.split(string)
+          .map { it.trim('\n') }
+          .drop(1) // TODO() - is drop 1 still needed?
```

Comment:
I just double checked, it seems so. The usage of split seems to grab the content before the `unrealeased` bit- so that's where the `drop` comes into play.